test_data/asm/pil_at_module_level.asm

```diff
@@ -0,0 +1,43 @@
+let x = 12;
+
+mod utils {
+    // Returns folder(...folder(folder(0, f(0)), f(1)) ..., f(length - 1))
+    let fold = |length, f, initial, folder| match length {
+        0 => initial,
+        _ => folder(fold(length - 1, f, initial, folder), f(length - 1))
+    };
+
+    /// creates the array [f(0), f(1), ..., f(length - 1)]
+    let make_array = |length, f| fold(length, f, [], |acc, e| acc + [e]);
+
+    /// returns f(0) + f(1) + ... + f(length - 1)
+    let sum = |length, f| fold(length, f, 0, |acc, e| acc + e);
+
+    use super::x as r;
+    let y = r;
+}
+
+mod R {
+    use super::x;
+    use super::utils::y;
+    use super::utils::sum;
+    use super::utils::make_array;
+
+    machine FullConstant {
+        degree 2;
+
+        let C = |i| match i % 2 {
+            0 => x,
+            1 => y,
+        };
+        col commit w[2];
+
+        // This and the next line is the same.
+        super::utils::sum(2, |i| w[i]) == 8;
+        sum(2, |i| w[i]) == 8;
+
+        // Try to see if we only clear local variables
+        // if they not already exist in the outer scope.
```

Comment:
```suggestion
        // if they don't already exist in the outer scope.
```

Reply:
wdym by "clear" here?

Reply:
`i` needs to be added to the list of local variables when entering the body of `|i| ...`. It needs to be cleared from that list again when we exit. There is a special situation where `i` is also a local variable of the outer function. In that case we of course do not clear.